py/serve.py

```diff
@@ -118,12 +118,12 @@ def show_dataset(name):
 @app.route('/dataset/<name>/download')
 def download_dataset(name):
     """Returns a pbtxt from the datasets table as an attachment."""
-    pbtxt = get_pbtxt(name)
-    data = io.BytesIO(pbtxt.encode('utf8'))
+    dataset = get_dataset(name)
+    data = io.BytesIO(dataset.SerializeToString(deterministic=True))
     return flask.send_file(data,
                            mimetype='application/protobuf',
                            as_attachment=True,
-                           attachment_filename=f'{name}.pbtxt')
+                           attachment_filename=f'{name}.pb')
```

Comment:
How complicated would it be to let people choose between pb and pbtxt when downloading? I think there is value in the ability to download the pbtxt in case you want to manually inspect it. Otherwise, users of the graphical editor would need to use the command line to convert the pb to pbtxt

Reply:
I think we'd need a separate button for each type (or something like a dropdown on the download button that gives both options), but that's totally reasonable.

Reply:
Done! (There are now two download buttons on the Dataset page.)